### PR TITLE
Convert license utils and constants to TS

### LINKED ISSUE #931: Include `utils/license.js` in `tsconfig.json`
## Description

Enable type checking on `utils/license.js`. This module has zero dependencies and is already type annotated but is not currently actually included in the type checking. Add it to the `tsconfig.json` and fix any errors that arise.

## Additional context

Part of an ongoing effort to add type checking to parts of the project that can be type checked.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #931

## Description

This PR converts the license constants and utils to TypeScript. In the process it also cleans up related files.

## Notable changes:

- Removed license abbreviations from translations as they are not translated. Leaving them in the `en.json` file can cause confusion such the as the 'BY' licence being translated to 'POR'.
- Adds 'CC' prefix in the filters sidebar, which is a UI/UX change that might need @panchovm's approval.

## Testing Instructions

There should be no runtime impact on the application. Can be verified by casually surfing through the pages.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by sarayourfriend — APPROVED** [review]
LGTM! I love moving that logic out of the template. Adding the CC prefix to all the license names feels a little redundant looking at it in the UI but it is technically more correct :thinking: Glad we have a designer who can make these decisions for us :stuck_out_tongue:

File: src/utils/license.ts [hunk 1]
```diff
@@ -1,47 +1,45 @@
...
  */
 export const getFullLicenseName = (
   license: License,
-  licenseVersion: LicenseVersion = '' // unknown version
-): LicenseName => {
-  if ((PUBLIC_DOMAIN_MARKS as ReadonlyArray<License>).includes(license)) {
-    const licenseUpper = license.toUpperCase() as Uppercase<PublicDomainMark>
-    return `${licenseUpper} ${licenseVersion}`
+  licenseVersion: LicenseVersion = '', // unknown version
+  i18n: IVueI18n | null = null
```
[the remark this answers is not in the document]
Comment (dhruvkb) [reply src/utils/license.ts:22]: This can be `null` if you're invoking the function from outside a Vue component (thus don't have access to the `i18n` instance) and are okay with seeing PDM as the license name.

File: src/utils/license.ts [hunk 2]
```diff
@@ -1,47 +1,45 @@
...
+): string | undefined => {
```
[the remark this answers is not in the document]
Comment (dhruvkb) [reply src/utils/license.ts]: This should never be `undefined`, you've caught a bug. I've removed that bit in a7d22dd.

File: src/utils/license.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dhruvkb) [reply src/utils/license.ts:22]: Ideally I updated all occurrences of this function to provide this argument, but the function should still be usable if `i18n` was unavailable.

File: src/layouts/default.vue [hunk 3]
```diff
@@ -110,7 +110,7 @@ export default embeddedPage
   .main {
     height: 100%;
     display: grid;
-    grid-template-columns: 1fr 316px;
+    grid-template-columns: 1fr 336px;
```
Comment (obulat) [inline src/layouts/default.vue:113]: What is the reason for this change? The sidebar is 316 px in the mockups.

**Review by obulat — APPROVED** [review]
Nice to have the domain knowledge about the licenses extracted and documented so well.
I left a note about the sidebar width, which is probably an unrelated change.

File: src/layouts/default.vue (hunk 3 above)
Comment (dhruvkb) [reply src/layouts/default.vue:113]: Adding CC to the license names caused the names to wrap in the sidebar. Adding a few pixels allow the names to fit in one line each.

**Comment by dhruvkb** [comment]
I'll merge this, tagging @panchovm to perform a design review and recommend any changes that might be needed.

**Comment by panchovm** [comment]
It looks great ✨ 

Thanks for adding the CC prefix. I will update the next release's mockups once we have the new set of pages in the Figma file.
